auto_submit/lib/configuration/repository_configuration_manager.dart

```diff
@@ -69,7 +69,7 @@ class RepositoryConfigurationManager {
     final RepositoryConfiguration globalRepositoryConfiguration = RepositoryConfiguration.fromYaml(orgLevelConfig);
 
     // Collect the default branch if it was not supplied.
-    if (globalRepositoryConfiguration.defaultBranch.isEmpty) {
+    if (globalRepositoryConfiguration.defaultBranch == 'default') {
       globalRepositoryConfiguration.defaultBranch = await githubService.getDefaultBranch(slug);
```

Comment:
Why not calling githubService.getDefaultBranch(slug) on repository_configuration?

Reply:
Also would we need to cache the results of githubService.getDefaultBranch(slug)?

Reply:
The reason is that you can actually set the default branch here and to avoid too many responsibilities the manager makes the call as it has the config and access to the Github service.

The results are cached so we are not making this call every time.

Reply:
In that case let's define the 'default' string as a shared constant to avoid issues when updating in one file but not the other.

Reply:
Ah good point!

Reply:
Done!